### PR TITLE
Add splitDirection property for PointPrimitive and Billboards

### PR DESCRIPTION
**Pull request by YunVlad**
# Description

Add the splitDirection property for PointPrimitive and Billboards.
Objects for which splitDirection is set as SplitDirection.NONE (by default) will be displayed to the right and left of Scene.splitPosition.
Objects for which splitDirection is set as SplitDirection.LEFT will be displayed to the left of Scene.splitPosition.
Objects for which splitDirection is set as SplitDirection.RIGHT will be displayed to the right of Scene.splitPosition.

This property is necessary for marking points of interest and subsequent analysis of changes over time.

# Author checklist

- [x] I have submitted a Contributor License Agreement
- [x] I have added my name to `CONTRIBUTORS.md`
- [x] I have updated `CHANGES.md` with a short summary of my change
- [x] I have added or updated unit tests to ensure consistent code coverage
- [x] I have update the inline documentation, and included code examples where relevant
- [x] I have performed a self-review of my code

### CODE REVIEW

**Comment by ggetz** [comment]
Awesome, thanks for the contribution @YunVlad! We'll review this shortly!

File: packages/engine/Source/Scene/PointPrimitiveCollection.js [hunk 1]
```diff
@@ -497,6 +500,12 @@ function createVAF(context, numberOfPointPrimitives, buffersUsage) {
         componentDatatype: ComponentDatatype.FLOAT,
         usage: buffersUsage[DISTANCE_DISPLAY_CONDITION_INDEX],
       },
+      {
+        index: attributeLocations.splitDirection,
```
Comment (ggetz) [inline packages/engine/Source/Scene/PointPrimitiveCollection.js:504]: To save on the number of attributes, which is limited, let's pack this single float component along with `distanceDisplayConditionAndDisableDepth` as a 4th component.

**Review by ggetz — COMMENTED** [review]
Awesome @YunVlad!

It would be great to include sample code for this in a Sandcastle example. Would you be open to either adding a new example, or extending an existing example such as [this one](https://sandcastle.cesium.com/?src=3D%20Tiles%20Compare.html) to showcase this feature?

File: CHANGES.md [hunk 2]
```diff
@@ -1,5 +1,7 @@
 # Change Log
 
+- Added SplitDirection property for display PointPrimitive relative to the `Scene.splitPosition`.
```
[the remark this answers is not in the document]
Comment (YunVlad) [reply CHANGES.md]: It was done. 
I also added an example in Sandcastle.

File: packages/engine/Source/Scene/PointPrimitiveCollection.js (hunk 1 above)
Comment (YunVlad) [reply packages/engine/Source/Scene/PointPrimitiveCollection.js]: Thank you for the comment. 
Fixed, added splitDirection as the 4th component in distanceDisplayConditionAndDisableDepth.

**Comment by ggetz** [comment]
Thanks for the updates @YunVlad!

I noticed one bug during my testing:

When adding a point with `viewer.entities.add`, setting `heightReference` to `HeightReference.CLAMP_TO_GROUND` causes the split property to stop working. [Here's a local Sandcastle example demonstrating the issue.](http://localhost:8080/Apps/Sandcastle/index.html#c=1VYNb+I4EP0rPvZWCnfU0O1Vp4O2ui5kS7Q0QUBbVUJiQzKAtcaOYgf6If772XFCk5a2265Op6MSJON5b8ZvxuMGnAmJVgTWEKNjxGCN2iBIssSXqc0aV4L0vc2Z9AmDeFyptsZszIIUGXHCpCgj+9rWj8mSSLKCNqcUAkk4szTQALAfhtb9mCFFIIhebBYZ2n4s1ZPPDvAs5ssOzGMAYe39eVhDB4fVmgYGnPK4uUXoNzywO+maiCiRHRKbuFunYcmMB85Zd5T6L4DMF3IAM4iBBbAFdMt23O6dnvcnI29yNvAuXBMrJMKfUuhAJBcjEIpfSD8lcZPlFGLc94bOyLm0J477xXGd0bWCbX5aiv1Pr2hx1XVG9hvUcD3X/t+K0WjU0KfG82Jc272ed/UGNXr2l/+yNcbMHEkMTCoZYKcwrx4WrFT5o4EzXVKFmyglUTIs+LqJZJxAzRh2yTb8ev25d2FnHhG5ATokdyr//UZm44mkaiq0d4DPBrbtlt2uSCgXTXSQWd9Yip8rxjvLgdDmSVFEAAxwlE84UxzTwIVWLu9OjcgXplBhogpKwnQWhzxIlqr+eA7SpqAfP986oZrIxsUM4lJKach+1iDoOG331Bfz2UyA7MFMonpui3ylVk6deaQVKmSz8FlIH18Nw0B1GBtGfgD2SmG7xskyvEYqChIt+QpOAy2Rws98KiBdmiXMSKLXLf2lE6iaziQzZP3yAKzm/RqDTGLWSgtiTrlOLwbqazcvTV5FydnUwQlzIfBN6wGxQyKErKcq/f6Iu4rqWdO+Lt7WSchbCphqvmP07dd7NadwA/1WTmLz8VuKeamSZYRy32ghs+JgFdphUSJPzTW71dfK1CvVQR95TVB7rpqj2wjSQzfpeFeurua/GKfvuO3uZDg6HYxM3zwTS1O/zHTuXQztybl3ab+Q8us0JqGc5p17z3v9R0S+6L9b4h8MY7Zku51qC1VqlaO0K09MNyP0N1lGPJYoiamFcV3CMlJ9D6I+TYLvIHEghFHCuH/I5tN9bijeSP5UcJpIaD0s6uZvosPGx4JN8qiJGtFNwTT1g+/zmCcs3Msuog/hgf4r+KzN7XFYAporQd9IpRB3e4SFcNNEf6nP1r55so/mQgla2k2QxELHh/VerES9gyJa/xzViwIehWSFSHi84/9kFFBfCLUyS2h6b44rW9UfYPkwPzmqK1vOWnzeulLuh4TNPZUx9W81ZLF/0jNGjPFRXb3uRkrO6dR/HOUf)

**Comment by YunVlad** [comment]
I found out why the [bug](https://github.com/CesiumGS/cesium/pull/11982#issuecomment-2140831603) occurs. Until the heightReference parameter is applied, points are rendered by PointPrimitiveCollection shaders (and splitDirection works as intended). However, after applying heightReference, another shader is clearly responsible for rendering (and, unfortunately, it does not work with splitDirection). But I can't figure out which one. Could you help with this question?

I would also like to note that PointPrimitive does not have a heightReference parameter.

**Comment by ggetz** [comment]
> However, after applying heightReference, another shader is clearly responsible for rendering (and, unfortunately, it does not work with splitDirection). But I can't figure out which one. Could you help with this question?

When clamped to ground, points are then represented by billboards rater than Point Primitives. I believe for this to work with the Entity API, you'll also need to implement splitDirection for billboard primitives, namely in `BillboardCollectionFS.glsl`.

**Comment by YunVlad** [comment]
Added the splitDirection property for Billboards. Completely by analogy with the one proposed above for Points.
Updated shaders, tests, and also updated the example for Sandcastle.

The [bug](http://localhost:8080/Apps/Sandcastle/index.html#c=1VYNb+I4EP0rPvZWCnfU0O1Vp4O2ui5kS7Q0QUBbVUJiQzKAtcaOYgf6If772XFCk5a2265Op6MSJON5b8ZvxuMGnAmJVgTWEKNjxGCN2iBIssSXqc0aV4L0vc2Z9AmDeFyptsZszIIUGXHCpCgj+9rWj8mSSLKCNqcUAkk4szTQALAfhtb9mCFFIIhebBYZ2n4s1ZPPDvAs5ssOzGMAYe39eVhDB4fVmgYGnPK4uUXoNzywO+maiCiRHRKbuFunYcmMB85Zd5T6L4DMF3IAM4iBBbAFdMt23O6dnvcnI29yNvAuXBMrJMKfUuhAJBcjEIpfSD8lcZPlFGLc94bOyLm0J477xXGd0bWCbX5aiv1Pr2hx1XVG9hvUcD3X/t+K0WjU0KfG82Jc272ed/UGNXr2l/+yNcbMHEkMTCoZYKcwrx4WrFT5o4EzXVKFmyglUTIs+LqJZJxAzRh2yTb8ev25d2FnHhG5ATokdyr//UZm44mkaiq0d4DPBrbtlt2uSCgXTXSQWd9Yip8rxjvLgdDmSVFEAAxwlE84UxzTwIVWLu9OjcgXplBhogpKwnQWhzxIlqr+eA7SpqAfP986oZrIxsUM4lJKach+1iDoOG331Bfz2UyA7MFMonpui3ylVk6deaQVKmSz8FlIH18Nw0B1GBtGfgD2SmG7xskyvEYqChIt+QpOAy2Rws98KiBdmiXMSKLXLf2lE6iaziQzZP3yAKzm/RqDTGLWSgtiTrlOLwbqazcvTV5FydnUwQlzIfBN6wGxQyKErKcq/f6Iu4rqWdO+Lt7WSchbCphqvmP07dd7NadwA/1WTmLz8VuKeamSZYRy32ghs+JgFdphUSJPzTW71dfK1CvVQR95TVB7rpqj2wjSQzfpeFeurua/GKfvuO3uZDg6HYxM3zwTS1O/zHTuXQztybl3ab+Q8us0JqGc5p17z3v9R0S+6L9b4h8MY7Zku51qC1VqlaO0K09MNyP0N1lGPJYoiamFcV3CMlJ9D6I+TYLvIHEghFHCuH/I5tN9bijeSP5UcJpIaD0s6uZvosPGx4JN8qiJGtFNwTT1g+/zmCcs3Msuog/hgf4r+KzN7XFYAporQd9IpRB3e4SFcNNEf6nP1r55so/mQgla2k2QxELHh/VerES9gyJa/xzViwIehWSFSHi84/9kFFBfCLUyS2h6b44rW9UfYPkwPzmqK1vOWnzeulLuh4TNPZUx9W81ZLF/0jNGjPFRXb3uRkrO6dR/HOUf) specified in the local example no longer occurs.

**Comment by ilyaly** [comment]
Hi there, is there any chance this PR will be merged anytime soon? This feature looks really cool!

**Comment by ggetz** [comment]
Thanks @YunVlad! Apologies for the delay! This code is looking good. I'm going to go ahead and push a few small updates to make this feature ready for the next release.

**Pull request merged**